specs/pvs/corla.pvs

```diff
@@ -2800,23 +2803,112 @@ BEGIN
+%%% asynchronously (our plan for a later deliverable). Asynchronous
+%%% updates happen via a call to the \texttt{refresh...} endpoints.
   upload_response: json_spec =
-    % legal values for status are:
-    % "too_late", "data_transmission_interrupted", "upload_sucessful",
-    % "checking_hash", "hash_wrong", "hash_verified", "parsing_data",
-    % "file_type_wrong", "data_parsed"
-    (: "status" :)
+%%% Legal values for \texttt{status} are: \texttt{too\_late},
+%%% \texttt{data\_\-transmission\_\-interrupted},
+%%% \texttt{upload\_\-sucessful}, \texttt{checking\_hash},
+%%% \texttt{hash\_wrong}, \texttt{hash\_verified},
+%%% \texttt{parsing\_\-data}, \texttt{file\_\-type\_\-wrong}, and
+%%% \texttt{data\_parsed}.
+    (: "ballot_manifest-status", "cvr-status" :)
+%%% There are several possible cases where a client can make an
+%%% inappropriate call to the server. Our client will be guarnateed to
+%%% never make sure a call, but we cannot presume that only our client
+%%% will be communicating with the server, as the server will be
+%%% publicly accessible on the internet. We call these behaviors of
+%%% the server ``exceptional behavior''. All responses to exceptional
+%%% behavior come in the form of HTTP client error (\texttt{3xx}) or
+%%% server error (\texttt{4xx}) response codes.
+  http_response_code: TYPE = {
+    OK_200,
+    BAD_REQUEST_400,
+    UNAUTHORIZED_401,
+    FORBIDDEN_403,
+    NOT_FOUND_404,
+    UNSUPPORTED_MEDIA_TYPE_415,
+    UNPROCESSABLE_ENTITY_422,
+    INTERNAL_SERVER_ERROR_500
+  }
+%%% The cases that we must cover are:
+%%% \begin{enumerate}
+%%%
+%%% \item The server is simply not up and running properly. For
+%%% example, if the database is not in a legal state. In these cases
+%%% the server will answer with an HTTP response code of
+%%% \texttt{INTERNAL\_SERVER\_ERROR\_500}.
+%%%
+%%% \item In instances where the server is asked to query for a
+%%% particular piece of data from the database and that does does not
+%%% exist, the server will respond with a \texttt{NOT\_FOUND\_404}. This
+%%% can happen, e.g., when a client uses an improper encoding for a
+%%% piece of data, such as a county ID that does not exist.
+%%%
+%%% \item A piece of data uploaded by a client (at this point in time,
+%%% either a ballot manifest, a CVR file, or an Audit CVR) is
+%%% malformed and cannot be interpreted. This situation can happen,
+%%% for example, if the Election Management System used to export the
+%%% file has a bug, if the file is improperly edited by someone, or if
+%%% there is a bug in our parser. In this situation, the server will
+%%% respond with a \texttt{UNPROCESSABLE\_ENTITY\_422}
+%%% response.
+%%%
+%%% \item Alternatively, if the file is of the wrong type (i.e., a PDF
+%%% was uploaded when the server expected a CSV file), or if the
+%%% uploaded file is too large, then the server will respond with a
+%%% \texttt{UNSUPPORTED\_MEDIA\_TYPE\_415} message.
```

Comment:
I was, so far, not attempting to detect the content type. I didn't intend to do so. It's not even clear to me what the best way of doing so would be (beyond trying to parse it as CSV). Of course if people end up uploading Zip files full of JSON, well, that's a different story.